query.py

```diff
@@ -437,9 +439,23 @@ def init_class(cls) -> None:
         cls._tree_processors = [t[1] for t in sorted(tree_procs, key=lambda x: -x[0])]
         cls._text_processors = [t[1] for t in sorted(text_procs, key=lambda x: -x[0])]
 
-        # Obtain query grammar fragments from the tree processors
+        # Query grammar plug-ins:
+        # Additional grammar fragments,
+        # which can be used in query module grammars
+        cls._query_grammar_additions = []
+        plugnames = modules_in_dir("queries/plugins")
+        for plug in plugnames:
+            try:
+                m = importlib.import_module(plug)
+                cls._query_grammar_additions.append(m)
+            except ImportError as e:
+                logging.error(
+                    "Error importing query plugin {0}: {1}".format(plug, e)
+                )
+
+        # Obtain query grammar fragments from the tree processors and query plugins
         grammar_fragments: List[str] = []
-        for processor in cls._tree_processors:
+        for processor in cls._tree_processors + cls._query_grammar_additions:
```

Comment:
Might be better to have the query grammar additions in front of the tree processors. I think the grammar parser handles forward references OK, but nevertheless it is logical to include them in that order.